language/move-lang/src/parser/syntax.rs

```diff
@@ -715,7 +728,19 @@ fn parse_exp<'input>(tokens: &mut Lexer<'input>) -> Result<Exp, Error> {
         }
         Tok::Return => {
             tokens.advance()?;
-            let e = Box::new(parse_exp(tokens)?);
+            let e = if at_end_of_exp(tokens) {
+                // If the return value expression is not present, construct a default
+                // unit value (with the same source location as the return keyword).
+                let default_value = spanned(
+                    tokens.file_name(),
+                    start_loc,
+                    tokens.previous_end_loc(),
+                    Exp_::Unit,
+                );
+                Box::new(default_value)
```

Comment:
Let's make `parser::ast::Exp_::Return` take an option. And then  apply this default_value fix expansion. 
It keeps the parser::ast a bit higher in fidelity, and keeps it inline with the other sort of "default" behavior syntax situations

Reply:
I pushed an updated patch that does this. How does it look now?